src/compiler/transformers/module/module.ts

```diff
@@ -820,6 +831,14 @@ namespace ts {
             let statements: Statement[];
             let variables: VariableDeclaration[];
             let expressions: Expression[];
+
+            const parseTreeNode = getParseTreeNode(node);

```

Comment:
Why do you need this check? Wouldn't setting `shouldAppendEsModuleMarker` to `true` in `createExportExpression` be sufficient? Basically, if I export *anything* other than via an `export=`, then I need to append `__esModule`. Would that be correct? Alternatively, always add it if you don't have an `export=` (e.g. `currentModuleInfo.exportEquals === false`).